docs/contributing/advanced_development_topics.rst

```diff
@@ -0,0 +1,156 @@
+
+- Include a gallery image.
+
+    Please also add the visualizer image to the gallery!
+
+Branching Convention
+--------------------
+
+The Yellowbrick repository is set up in a typical production/release/development cycle as described in "`A Successful Git Branching Model <http://nvie.com/posts/a-successful-git-branching-model/>`_." The primary working branch is the ``develop`` branch. This should be the branch that you are working on and from, since this has all the latest code. The ``master`` branch contains the latest stable version and release_, which is pushed to PyPI_. No one but core contributors will generally push to master.
+
+.. note:: All pull requests should be into the ``yellowbrick/develop`` branch from your forked repository.
+
+You can work directly in your fork and create a pull request from your fork's develop branch into ours. We also recommend setting up an ``upstream`` remote so that you can easily pull the latest development changes from the main Yellowbrick repository (see `configuring a remote for a fork <https://help.github.com/articles/configuring-a-remote-for-a-fork/>`_). You can do that as follows::
+
+    $ git remote add upstream https://github.com/DistrictDataLabs/yellowbrick.git
+    $ git remote -v
+    origin    https://github.com/YOUR_USERNAME/YOUR_FORK.git (fetch)
+    origin    https://github.com/YOUR_USERNAME/YOUR_FORK.git (push)
+    upstream  https://github.com/DistrictDataLabs/yellowbrick.git (fetch)
+    upstream  https://github.com/DistrictDataLabs/yellowbrick.git (push)
+
+When you're ready, request a code review for your pull request. Then, when reviewed and approved, you can merge your fork into our main branch. Make sure to use the "Squash and Merge" option in order to create a Git history that is understandable.
+
+.. note:: When merging a pull request, use the "squash and merge" option.
+
+Core contributors have write access to the repository. In order to reduce the number of merges (and merge conflicts) we recommend that you utilize a feature branch off of develop to do intermediate work in::
+
+    $ git checkout -b feature-myfeature develop
+
+Once you are done working (and everything is tested) merge your feature into develop.::
+
+    $ git checkout develop
+    $ git merge --no-ff feature-myfeature
+    $ git branch -d feature-myfeature
+    $ git push origin develop
+
+Head back to Waffle and checkout another issue!
+
+Releases
+--------
+
+When ready to create a new release we branch off of develop as follows::
+
+    $ git checkout -b release-x.x
+
+This creates a release branch for version x.x. At this point do the version bump by modifying ``version.py`` and the test version in ``tests/__init__.py``. Make sure all tests pass for the release and that the documentation is up to date. There may be style changes or deployment options that have to be done at this phase in the release branch. At this phase you'll also modify the ``changelog`` with the features and changes in the release.
```

Comment:
Do we have instructions somewhere for building the documentation that we can reference here? Feel like it's a big thing to make sure that docs are built and working before the release